### PR TITLE
Add sub categories to job sections

### LINKED ISSUE #1072: Recategorize and sort forward models in documentation
Forward models are filling up in the documentation. Some of them should possibly be recategorized and there should be sorting within each category.  Categories are set in multiple repositories out of control of ERT itself, and should be a coordinated effort.

image

### PR DESCRIPTION
**Pull request by sondreso**
**Issue**
Resolves #1072 

**Approach**
Add sub categories to forward model jobs section and workflow jobs section

image

### CODE REVIEW

File: ert_shared/_doc_utils/ert_jobs.py [hunk 1]
```diff
@@ -138,11 +139,15 @@ def _divide_into_categories(jobs):
                 _ErtDocumentation._CATEGORY_DEFAULT,
             )
 
-            main_category = category.split(".")[0]
-            if main_category not in categories:
-                categories[main_category] = list()
+            split_categories = category.split(".")
+            if len(split_categories) > 1:
+                main_category, sub_category = split_categories[0:2]
```
Comment (markusdregi) [inline ert_shared/_doc_utils/ert_jobs.py:144]: Are we not going to warn or error if someone injected a third category?

File: ert_shared/_doc_utils/ert_jobs.py [hunk 2]
```diff
@@ -185,18 +190,35 @@ def _generate_job_documentation(self, jobs, section_id, title):
             category_section_node = _create_section_with_title(
                 section_id=category + "-category", title=category.capitalize()
             )
+            job_sub_categories = job_categories[category]
+            print("Start", category)
```
Comment (markusdregi) [inline ert_shared/_doc_utils/ert_jobs.py:194]: Traces of debugging?

File: ert_shared/_doc_utils/ert_jobs.py [hunk 3]
```diff
@@ -185,18 +190,35 @@ def _generate_job_documentation(self, jobs, section_id, title):
...
+            for subcategory_index, subcategory in enumerate(
+                sorted(job_sub_categories.keys())
+            ):
+                print("Start sub", subcategory)
```
Comment (markusdregi) [inline ert_shared/_doc_utils/ert_jobs.py:198]: Same...

File: ert_shared/_doc_utils/ert_jobs.py [hunk 4]
```diff
@@ -185,18 +190,35 @@ def _generate_job_documentation(self, jobs, section_id, title):
             category_section_node = _create_section_with_title(
                 section_id=category + "-category", title=category.capitalize()
             )
+            job_sub_categories = job_categories[category]
+            print("Start", category)
+            for subcategory_index, subcategory in enumerate(
```
Comment (markusdregi) [inline ert_shared/_doc_utils/ert_jobs.py:195]: I'm suspecting all of this code would read nicer if you are able to abbreviate the two variable names

File: ert_shared/_doc_utils/ert_jobs.py (hunk 1 above)
Comment (sondreso) [reply ert_shared/_doc_utils/ert_jobs.py:144]: The idea was the the categories are hierarchical, so one could very well inject a third of fourth sub-category to make it even more specific. However, the grouping in the documentation will only group by the two top level categories :slightly_smiling_face:

**Review by markusdregi — APPROVED** [review]
Please squash before you merge 👏

**Pull request merged**